src/profile-logic/symbolication.js

```diff
@@ -515,34 +504,30 @@ export function applySymbolicationStep(
     funcTable.name[funcIndex] = symbolStringIndex;
   }
 
-  // Now we have a canonical func for every funcAddress, so we have enough
-  // information to build the oldFuncToNewFuncMap.
-  // If (oldFunc, newFunc) is in oldFuncToNewFuncMap, this means that all
-  // frames that used to belong to oldFunc or newFunc have been resolved to
-  // the same funcAddress, and that newFunc has been chosen as the canonical
-  // func for that funcAddress.
-  for (const [oldFunc, funcAddress] of oldFuncToFuncAddressMap) {
-    const newFunc = funcAddressToCanonicalFuncIndexMap.get(funcAddress);
-    if (newFunc === undefined) {
-      throw new Error(
-        'Impossible, all funcAddresses have a canonical func index at this point.'
-      );
-    }
-    if (oldFuncToFuncAddressMap.get(newFunc) === funcAddress) {
-      oldFuncToNewFuncMap.set(oldFunc, newFunc);
-    }
-  }
-
   // Make a new frameTable with the updated frame -> func assignments.
   const newFrameTableFuncColumn: Array<IndexIntoFuncTable> = oldFrameTable.func.slice();
+  const oldFuncToNewFuncEntries = [];
   for (const [frameIndex, funcAddress] of frameToFuncAddressMap) {
+    const oldFuncIndex = oldFrameTable.func[frameIndex];
     const funcIndex = funcAddressToCanonicalFuncIndexMap.get(funcAddress);
     if (funcIndex === undefined) {
       throw new Error(
         'Impossible, all funcAddresses have a canonical func index at this point.'
       );
     }
     newFrameTableFuncColumn[frameIndex] = funcIndex;
+    oldFuncToNewFuncEntries.push([oldFuncIndex, funcIndex]);
+  }
+
+  // Build oldFuncToNewFuncMapForThisLib.
+  // If (oldFunc, newFunc) is in oldFuncToNewFuncMapForThisLib, this means
+  // that all frames that used to belong to oldFunc have been resolved to
+  // the same func newFunc.
+  const oldFuncToNewFuncMapForThisLib = makeConsensusMap(
```

Comment:
Ah, I didn't know that we had `makeConsensusMap`. Looks interesting. Even though it's a niche, I wonder if it would be useful for some other areas. Might be good to move it out to a different file in the future. (not for this PR ofc.)